In [10]:
import pandas as pd
import matplotlib.pyplot as plt

In [11]:
df = pd.read_csv(filepath_or_buffer= "Extract_Dataset.csv") 
df

Unnamed: 0                     App Name  \
0              0                        Untap   
1              1                Green Meadows   
2              2                    YG SELECT   
3              3                 Vinca Wealth   
4              4                Drink recipes   
...          ...                          ...   
9995        9995  Deep Memorial Public School   
9996        9996                Anchored Sins   
9997        9997                 설운도 트로트 노래모음   
9998        9998                          Coq   
9999        9999                  Color Kraft   

                                    App Id       Category  Rating  \
0                                 in.untap  Entertainment     3.9   
1     com.ooweboowebengineers.greenmeadows      Lifestyle     0.0   
2             com.makeshop.powerapp.ygnext       Shopping     4.3   
3                           com.bag4wealth        Finance     5.0   
4                       com.drinks.recipes   Food & Drink     4.3   
...                                    ...            ...     ...   
9995                      com.edunext.dmps      Education     4.3   
9996                com.goc.anchoredsins3D   Role Playing     NaN   
9997                  korea.singer.sulundo  Music & Audio     5.0   
9998                   fr.enfantdoudou.coq  Entertainment     0.0   
9999                 com.ekraft.colorkraft         Casual     4.9   

      Rating Count  Installs  Minimum Installs  Maximum Installs  Free  ...  \
0             68.0   10,000+           10000.0             10291  True  ...   
1              0.0       50+              50.0                90  True  ...   
2            918.0  100,000+          100000.0            135038  True  ...   
3              6.0       50+              50.0                53  True  ...   
4            830.0  100,000+          100000.0            142498  True  ...   
...            ...       ...               ...               ...   ...  ...   
9995         142.0    1,000+            1000.0              2490  True  ...   
9996           NaN        5+               5.0                 8  True  ...   
9997           9.0    1,000+            1000.0              1035  True  ...   
9998           0.0      500+             500.0               522  True  ...   
9999          19.0      500+             500.0               656  True  ...   

                             Developer Website            Developer Email  \
0                             https://untap.in             hello@untap.in   
1                         http://ooweboo.co.za          ray@ooweboo.co.za   
2                       http://www.ygeshop.com     app.ygselect@gmail.com   
3                       https://bag4wealth.com     acmatics.app@gmail.com   
4                     http://cookwithlove.biz/   andrei.nazarco@gmail.com   
...                                        ...                        ...   
9995        http://www.edunexttechnologies.com      edunexttech@gmail.com   
9996  https://www.gearoverclocked.com/projects  gearoverclocked@gmail.com   
9997                       http://israng.co.kr      parkhanye28@gmail.com   
9998                                       NaN          pridgua@gmail.com   
9999                       http://ekraft.co.in   dakshpandyaweb@gmail.com   

          Released  Last Updated Content Rating  \
0      Nov 2, 2020  Nov 02, 2020       Everyone   
1     May 29, 2017  May 29, 2017       Everyone   
2     Jan 20, 2016  May 12, 2021       Everyone   
3     Jun 30, 2020  May 11, 2021       Everyone   
4     May 20, 2014  Jul 13, 2020       Everyone   
...            ...           ...            ...   
9995   Mar 4, 2016  Aug 26, 2020       Everyone   
9996           NaN  Jan 03, 2020           Teen   
9997  Apr 23, 2020  Feb 03, 2021           Teen   
9998  Apr 16, 2020  Apr 16, 2020       Everyone   
9999  Apr 24, 2015  Dec 23, 2015       Everyone   

                                         Privacy Policy Ad Supported  \
0                            

In [12]:
df = df.loc[:, ~df.columns.str.contains('^Unnamed')]
df

App Name                                App Id  \
0                           Untap                              in.untap   
1                   Green Meadows  com.ooweboowebengineers.greenmeadows   
2                       YG SELECT          com.makeshop.powerapp.ygnext   
3                    Vinca Wealth                        com.bag4wealth   
4                   Drink recipes                    com.drinks.recipes   
...                           ...                                   ...   
9995  Deep Memorial Public School                      com.edunext.dmps   
9996                Anchored Sins                com.goc.anchoredsins3D   
9997                 설운도 트로트 노래모음                  korea.singer.sulundo   
9998                          Coq                   fr.enfantdoudou.coq   
9999                  Color Kraft                 com.ekraft.colorkraft   

           Category  Rating  Rating Count  Installs  Minimum Installs  \
0     Entertainment     3.9          68.0   10,000+           10000.0   
1         Lifestyle     0.0           0.0       50+              50.0   
2          Shopping     4.3         918.0  100,000+          100000.0   
3           Finance     5.0           6.0       50+              50.0   
4      Food & Drink     4.3         830.0  100,000+          100000.0   
...             ...     ...           ...       ...               ...   
9995      Education     4.3         142.0    1,000+            1000.0   
9996   Role Playing     NaN           NaN        5+               5.0   
9997  Music & Audio     5.0           9.0    1,000+            1000.0   
9998  Entertainment     0.0           0.0      500+             500.0   
9999         Casual     4.9          19.0      500+             500.0   

      Maximum Installs  Free  Price  ...  \
0                10291  True    0.0  ...   
1                   90  True    0.0  ...   
2               135038  True    0.0  ...   
3                   53  True    0.0  ...   
4               142498  True    0.0  ...   
...                ...   ...    ...  ...   
9995              2490  True    0.0  ...   
9996                 8  True    0.0  ...   
9997              1035  True    0.0  ...   
9998               522  True    0.0  ...   
9999               656  True    0.0  ...   

                             Developer Website            Developer Email  \
0                             https://untap.in             hello@untap.in   
1                         http://ooweboo.co.za          ray@ooweboo.co.za   
2                       http://www.ygeshop.com     app.ygselect@gmail.com   
3                       https://bag4wealth.com     acmatics.app@gmail.com   
4                     http://cookwithlove.biz/   andrei.nazarco@gmail.com   
...                                        ...                        ...   
9995        http://www.edunexttechnologies.com      edunexttech@gmail.com   
9996  https://www.gearoverclocked.com/projects  gearoverclocked@gmail.com   
9997                       http://israng.co.kr      parkhanye28@gmail.com   
9998                                       NaN          pridgua@gmail.com   
9999                       http://ekraft.co.in   dakshpandyaweb@gmail.com   

          Released  Last Updated Content Rating  \
0      Nov 2, 2020  Nov 02, 2020       Everyone   
1     May 29, 2017  May 29, 2017       Everyone   
2     Jan 20, 2016  May 12, 2021       Everyone   
3     Jun 30, 2020  May 11, 2021       Everyone   
4     May 20, 2014  Jul 13, 2020       Everyone   
...            ...           ...            ...   
9995   Mar 4, 2016  Aug 26, 2020       Everyone   
9996           NaN  Jan 03, 2020           Teen   
9997  Apr 23, 2020  Feb 03, 2021           Teen   
9998  Apr 16, 2020  Apr 16, 2020       Everyone   
9999  Apr 24, 2015  Dec 23, 2015       Everyone   

                                         Privacy Policy Ad Supported  \
0                                                   NaN        False   
1     http://appmc2.net/privacy?comp

### Step 1: Check datatypes and format

In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 24 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   App Name           10000 non-null  object 
 1   App Id             10000 non-null  object 
 2   Category           10000 non-null  object 
 3   Rating             9906 non-null   float64
 4   Rating Count       9906 non-null   float64
 5   Installs           9999 non-null   object 
 6   Minimum Installs   9999 non-null   float64
 7   Maximum Installs   10000 non-null  int64  
 8   Free               10000 non-null  bool   
 9   Price              10000 non-null  float64
 10  Currency           9999 non-null   object 
 11  Size               9999 non-null   object 
 12  Minimum Android    9969 non-null   object 
 13  Developer Id       10000 non-null  object 
 14  Developer Website  6628 non-null   object 
 15  Developer Email    10000 non-null  object 
 16  Released           9694

We can that see there are a mixture of numerical and categorical columns. There is lots of "object" type which corresponds to categorical data. We can also see that there is boolean data that might be some binary type columns which could be True/False or 0/1 which are captured as "yes"/"no". Finally, we see that there are some columns missing data in Rating, Rating Count ,Installs, Minimum Installs, Currency, Size, Minimum Android, Developer Website (this columns has the more null value compare to the other columns), Released and Privacy Policy.

### Checking for duplicate data

In [22]:
df.duplicated().sum()

0

In [23]:
df.T.duplicated().sum()

0

According to this command there is no ducplicate information

### Dealing with missing data

In [24]:
df.isna().sum()

App Name                0
App Id                  0
Category                0
Rating                 94
Rating Count           94
Installs                1
Minimum Installs        1
Maximum Installs        0
Free                    0
Price                   0
Currency                1
Size                    1
Minimum Android        31
Developer Id            0
Developer Website    3372
Developer Email         0
Released              306
Last Updated            0
Content Rating          0
Privacy Policy       1746
Ad Supported            0
In App Purchases        0
Editors Choice          0
Scraped Time            0
dtype: int64

In [25]:
df.isna().sum()/df.shape[0] * 100

App Name              0.00
App Id                0.00
Category              0.00
Rating                0.94
Rating Count          0.94
Installs              0.01
Minimum Installs      0.01
Maximum Installs      0.00
Free                  0.00
Price                 0.00
Currency              0.01
Size                  0.01
Minimum Android       0.31
Developer Id          0.00
Developer Website    33.72
Developer Email       0.00
Released              3.06
Last Updated          0.00
Content Rating        0.00
Privacy Policy       17.46
Ad Supported          0.00
In App Purchases      0.00
Editors Choice        0.00
Scraped Time          0.00
dtype: float64

we can see that 0.9% of values are missing for the Rating and Rating Count columns, 0.01% of values are missing for the Installs, Minimum Installs, Currency, and Size columns, 0.31% of values are missing for the Minimum Android column, 33.72% of values are missing for the Developer Website column, 3.06% of values are missing for the Released column and 17.46% of values are missing for the Privacy Policy column."